kotlinx-coroutines-core/common/src/Timeout.kt

```diff
@@ -151,7 +151,12 @@ private fun <U, T: U> setupTimeout(
 private class TimeoutCoroutine<U, in T: U>(
     @JvmField val time: Long,
     uCont: Continuation<U> // unintercepted continuation
-) : ScopeCoroutine<T>(uCont.context, uCont), Runnable {
+) : ScopeCoroutine<T>(uCont.context, uCont, false), Runnable {
+    init {
+        // Kludge for native
+        if (!isReuseSupportedInPlatform()) initParentForNativeUndispatchedCoroutine()
```

Comment:
```
if (!isReuseSupportedInPlatform()) initParentForNativeUndispatchedCoroutine()
```
Personally, I prefer to make this as a `initParentJob` method in the superclass for maintainability